app/src/routes/v1/bucket.js

```diff
@@ -0,0 +1,41 @@
+const routes = require('express').Router();
+
+const { bucketController } = require('../../controllers');
+const { requireDb, requireSomeAuth } = require('../../middleware/featureToggle');
+const { checkAppMode } = require('../../middleware/authorization');
+
+routes.use(checkAppMode);
+routes.use(requireDb);
+routes.use(requireSomeAuth);
+
+/** Creates a bucket */
+routes.put('/', (req, res, next) => {
+  bucketController.createBucket(req, res, next);
+});
+
+/** Search for buckets */
+routes.get('/', (req, res, next) => {
+  bucketController.searchBuckets(req, res, next);
+});
+
+/** Returns bucket headers */
+routes.head('/:bucketId', (req, res, next) => {
+  bucketController.headBucket(req, res, next);
+});
+
+/** Returns a bucket */
+routes.get('/:bucketId', (req, res, next) => {
+  bucketController.readBucket(req, res, next);
+});
+
+/** Updates a bucket */
+routes.patch('/:bucketId', (req, res, next) => {
```

Comment:
i guess the permissions work your doing will be adding or extending the hasPermission type middleware stuff and adding it to these routes

Reply:
Correct. I still need to work through the XOR cascade logic for determining permissions between a bucket and object; logic refactor still TBD.